theano/sandbox/cuda/opt.py

```diff
@@ -710,6 +710,34 @@ def local_gpu_solve(node):
 
 
 @register_opt()
+@local_optimizer([gpu_from_host, slinalg.Cholesky])
+def local_gpu_cholesky(node):
+    """
+    gpu_from_host(CpuCholesky) -> GpuCholesky(gpu_from_host)
+
+    CpuCholesky(host_from_gpu) -> host_from_gpu(GpuCholesky)
+
+    """
+    if isinstance(node.op, GpuFromHost):
+        host_input = node.inputs[0]
+        if (host_input.owner and
+            isinstance(host_input.owner.op,
+                       slinalg.Cholesky)):
+            A = host_input.owner.inputs[0]
+            lower = host_input.owner.op.lower
+            return [gpu_cholesky(as_cuda_ndarray_variable(A), lower)]
+
+    if isinstance(node.op, slinalg.Cholesky):
+        if any([i.owner and isinstance(i.owner.op, HostFromGpu)
+                for i in node.inputs]):
```

Comment:
As the op always have only 1 input, you can remove the list comprehension and do the check only on node.inputs[0]. This make the code more clear.